src/Middleware/Rewrite/src/RedirectToRootFromWwwRule.cs

```diff
@@ -0,0 +1,83 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Http.Extensions;
+using Microsoft.AspNetCore.Rewrite.Logging;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.Rewrite
+{
+    internal class RedirectToRootFromWwwRule : IRule
+    {
+        public readonly int _statusCode;
+        public readonly string[] _domains;
+
+        public RedirectToRootFromWwwRule(int statusCode)
```

Comment:
I think we should rename this to RedirectToNonWwwRule. To me, root isn't very clear.